synapse/handlers/room_list.py

```diff
@@ -423,20 +432,21 @@ async def _get_remote_list_cached(
         include_all_networks: bool = False,
         third_party_instance_id: Optional[str] = None,
     ) -> JsonDict:
+        """Wrapper around FederationClient.get_public_rooms that caches the
```

Comment:
I've got to wonder if caching a RequestSendFailed is the right thing to do, but that's a separate problem. :man_shrugging:

Reply:
:man_shrugging: